test/core/tsi/s2a/s2a_test_util.cc

```diff
@@ -323,3 +219,102 @@ bool check_record_empty_plaintext(uint16_t ciphersuite,
   }
   return true;
 }
+
+void random_array(uint8_t* bytes, size_t length) {
+  if (length == 0) return;
+  GPR_ASSERT(bytes != nullptr);
+  for (size_t i = 0; i < length; i++) {
+    bytes[i] = static_cast<uint8_t>(rand() % 256);
+  }
+}
+
+void send_random_message(size_t message_size, s2a_crypter* out_crypter,
```

Comment:
I prefer not having randomized test input/vectors. The reason is because if there's a failure, you want to be able to reproduce it in the exact way how it happened. If we use random inputs, reproducing is harder.

What do you think about passing the message to this function?

Reply:
In the most recent commit, I've taken out the randomized tests and replaced them with test messages placed in a `s2a_test_data.h` header file. If you agree with the setup, I'll add to the list of test messages (in particular, add some long ones) and I will also add test keys.

Reply:
SG, the current version of the code LGTM.